src/range.cr

```diff
@@ -305,4 +305,46 @@ struct Range(B, E)
       self
     end
   end
+
+  #Perform binary search
+  #Based on Marc-André Lafortune's Ruby backports implementation, ported by T. Yamada
+  def bsearch
+    from = self.begin
+    to   = self.end
+    unless from.is_a?(Int) && to.is_a?(Int)
+      # Float support is currently dropped
+      raise "can't do binary search for #{from.class}"
+    end
+
+    midpoint = 0 # placeholder
+    convert = ->{ midpoint }
```

Comment:
What's the use of this proc? It forms a closure and it's slow (allocates heap memory).